### PR TITLE
[Anagram] - Add optimized method to detect Anagram.

### PR DESCRIPTION
**Pull request by ivandrofly**

### CODE REVIEW

**Comment by ivandrofly** [comment]
what do you think about using visual-studio is unit-test framework instead testing using "Debug.Assert"...? /cc @aalhour

File: Algorithms/Strings/Permutations.cs [hunk 1]
```diff
@@ -66,32 +66,28 @@ public static HashSet<string> ComputeDistinct(string Source)
...
+        public static bool IsAnargram(string source, string other)
         {
-            if (string.IsNullOrEmpty(Source) || string.IsNullOrEmpty(Other))
+            if (string.IsNullOrEmpty(source) || string.IsNullOrEmpty(other))
                 return false;
-            else if (Source.Length != Other.Length)
+            else if (source.Length != other.Length)
                 return false;
-            else if (Source == Other)
+            else if (source.Equals(other, StringComparison.Ordinal))
```
Comment (ivandrofly) [inline Algorithms/Strings/Permutations.cs]: `StringComparison.Ordinal` in order to avoid issue with turkish...

**Comment by aalhour** [comment]
Hello @ivandrofly,

Mentioning a Unit Testing framework brings music to my ears, when I originally developed this library, I didn't realise it will become a big project, but now that it is, it makes total sense of implementing tests in a unit-testing framework. Which one would you recommend?

Whatever your recommendation will be, please make sure it is available for .NET Core and Mono, I am developing this project on my Mac laptop.

Cheers,
Ahmad

**Comment by ivandrofly** [comment]
> Speaking of which, would you be willing to refactor the existing tests to use the new Unit Testing framework?

Sure :)

> Whatever your recommendation will be, please make sure it is available for .NET Core and Mono, I am developing this project on my Mac laptop.

I'll go with XUnit framework seems to work with MacOS

**Comment by aalhour** [comment]
This is awesome! Shall we merge this PR and then you will submit another one for the tests?

By the way, we should write an issue to refactor the test, so that we can exchange ideas about it before your PR is ready.

**Comment by ivandrofly** [comment]
> This is awesome! Shall we merge this PR and then you will submit another one for the tests?

Yeah this one can be merged.

> By the way, we should write an issue to refactor the test, so that we can exchange ideas about it before your PR is ready.

Sounds good :)

**Pull request merged**